### DISCUSSION TITLE
Async event handlers

### DISCUSSION THREAD

**Discussion by luc122c** [nuxtrdev/nuxtr-vscode#31] [Ideas]
When creating a new API route, the default template is:
```ts
export default defineEventHandler((event) => {
  return 'Hello test'
})
```
It would be great if the function was async, or if there was an option for the default function to be async. 

```ts
export default defineEventHandler(async (event) => {
  return 'Hello test'
})
```

**Comment by adhamfarrag** [comment]
Thanks for your feedback @luc122c. 

We're exploring better ways to inject snippets in new files with a typical experience to user snippets and `async` position will be provided. 

We're trying to make Nuxtr easy to configure and one of the ways is to avoid introducing new settings/options that can be handled otherwise. Will keep your idea in mind if we couldn't achieve our feature the way we aim for and directly add `async` to the snippet. 

From using Nuxtr so far, is there anything else missing or you don't like about it ?

**Reply by luc122c** [reply]
My only other gripe is that it seems to re-order my plugins in nuxt.config 😄 Apart from that, it's wonderful, and has really helped speed up my Nuxt development.

**Reply by adhamfarrag** [reply]
@luc122c We released V 0.1.11 with a fix to remove the plugins watcher. Please update it snd this issue is gone for good. :)

**Reply by luc122c** [reply]
Thanks @adhamfarrag keep up the great work! :)

**Reply by adhamfarrag** [reply]
Hey @luc122c, async has been added to the template.. :) 
Please, upgrade to Nuxtr 0.2.0!

**Reply by luc122c** [reply]
Thanks so much @adhamfarrag :)
